plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/create-product-attributes.spec.js

```diff
@@ -23,109 +37,130 @@ test.describe( 'Add product attributes', () => {
 	} );
 
 	test( 'can add custom product attributes', async ( { page } ) => {
-		await test.step(
-			`Open "Edit product" page of product id ${ productId }`,
-			async () => {
-				await page.goto(
-					`/wp-admin/post.php?post=${ productId }&action=edit`
-				);
-			}
+		const textbox_attributeName =
+			page.getByPlaceholder( 'f.e. size or color' );
+		const textbox_attributeValues = page.getByPlaceholder(
+			'Enter options for customers to choose from'
 		);
+		const checkbox_visible = page
+			.getByText( 'Visible on the product page' )
+			.getByRole( 'checkbox' );
+		const checkbox_variations = page
+			.getByText( 'Used for variations' )
+			.getByRole( 'checkbox' );
+
+		await test.step( `Open "Edit product" page of product id ${ productId }`, async () => {
+			await page.goto(
+				`/wp-admin/post.php?post=${ productId }&action=edit`
+			);
+		} );
 
-		await test.step( 'Go to the "Attributes" tab.', async () => {
-			const attributesTab = page
-				.locator( '.attribute_tab' )
-				.getByRole( 'link', { name: 'Attributes' } );
+		await step_goToAttributesTab( page );
+
+		for ( let i = 0; i < productAttributes.length; i++ ) {
+			const attribute = productAttributes[ i ];
+			const attributeName = attribute.name;
+			const attributeValues = attribute.options.join( ' | ' );
+
+			if ( i > 0 ) {
+				await test.step( "Click 'Add new'.", async () => {
+					await page
+						.getByRole( 'button', { name: 'Add new' } )
+						.click();
+
+					await page
+						.getByRole( 'heading', {
+							name: 'New attribute',
+						} )
+						.waitFor();
```

Comment:
Are we waiting for this to be visible?  Might be better to be explicit about it: https://playwright.dev/docs/testing-library#replacing-waitfor

Reply:
Oh, thank you very much! Yeah it's much better to be more explicit with the visibility check. Done on 3d923b13e7df08ba2c6b1bbd2a47b0b7f4e2d77a. And thanks, I learned something new 😄